websub-ballerina/src/websub/tests/unit-tests/test_discovery_response_parsing.bal

```diff
@@ -0,0 +1,226 @@
+    response.addHeader("Link", "<" + HUB_ONE + ">; rel=\"hub\", <" + HUB_TWO + ">; rel=\"hub\", <" + HUB_THREE +
+            ">; rel=\"hub\", <" + TOPIC_ONE + ">; rel=\"self\"");
+
+    var results = extractTopicAndHubUrls(response);
+    if (results is [string, string[]]) {
+        string topic = results[0];
+        string[] hubs = results[1];
+        test:assertEquals(topic, TOPIC_ONE, msg = "incorrect topic extraction from discovery response");
+        test:assertEquals(hubs.length(), 3, msg = "incorrect no. of hubs extracted from discovery response");
+        test:assertEquals(hubs[0], HUB_ONE, msg = "incorrect hub extraction from discovery response");
+        test:assertEquals(hubs[1], HUB_TWO, msg = "incorrect hub extraction from discovery response");
+        test:assertEquals(hubs[2], HUB_THREE, msg = "incorrect hub extraction from discovery response");
+    } else {
+        string? errMsg = results.message();
+        test:assertFail(msg = errMsg is string ? errMsg : "error in extractTopicAndHubUrls");
+    }
+}
+
+@test:Config {
+    dependsOn: ["testTopicAndMultipleHubsAsSingleLinkHeader"]
+}
+function testTopicAndMultipleHubsAsMultipleLinkHeaders() {
+    http:Response response = new;
+    response.addHeader("Link", "<" + HUB_ONE + ">; rel=\"hub\"");
+    response.addHeader("Link", "<" + TOPIC_ONE + ">; rel=\"self\"");
+    response.addHeader("Link", "<" + HUB_TWO + ">; rel=\"hub\"");
+    response.addHeader("Link", "<" + HUB_THREE + ">; rel=\"hub\"");
+
+    var results = extractTopicAndHubUrls(response);
+    if (results is [string, string[]]) {
+        string topic = results[0];
+        string[] hubs = results[1];
+        test:assertEquals(topic, TOPIC_ONE, msg = "incorrect topic extraction from discovery response");
+        test:assertEquals(hubs.length(), 3, msg = "incorrect no. of hubs extracted from discovery response");
+        test:assertEquals(hubs[0], HUB_ONE, msg = "incorrect hub extraction from discovery response");
+        test:assertEquals(hubs[1], HUB_TWO, msg = "incorrect hub extraction from discovery response");
+        test:assertEquals(hubs[2], HUB_THREE, msg = "incorrect hub extraction from discovery response");
+    } else {
+        string? errMsg = results.message();
+        test:assertFail(msg = errMsg is string ? errMsg : "error in extractTopicAndHubUrls");
+    }
+}
+
+@test:Config {
+    dependsOn: ["testTopicAndMultipleHubsAsMultipleLinkHeaders"]
+}
+function testMissingTopicWithSingleLinkHeader() {
+    http:Response response = new;
+    response.addHeader("Link", "<" + HUB_ONE + ">; rel=\"hub\", <" + TOPIC_ONE + ">; rel=\"not_self\"");
+    var results = extractTopicAndHubUrls(response);
+    if (results is error) {
+        string? errMsg = results.message();
+        string expectedErrMsg = "Hub and/or Topic URL(s) not identified in link header of discovery response";
+        test:assertEquals(errMsg, expectedErrMsg, msg = "invalid error message on unavailable topic");
```

Comment:
Here also we can directly use results.message() as actual message. We may need to fix other places as well
```suggestion
        test:assertEquals(results.message(), expectedErrMsg, msg = "invalid error message on unavailable topic");
```

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-websub/pull/1/commits/7defc19195f199d3899556e82f9a35eb1ebf4bd5